chanfitness/chaneventstore_test.go

```diff
@@ -0,0 +1,460 @@
+package chanfitness
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/channelnotifier"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/peernotifier"
+	"github.com/lightningnetwork/lnd/subscribe"
+)
+
+// TestStartStore tests the mechanics of starting up the store and shutting
+// it down. It does not test the mechanics of consuming events because mocking
+// subscribe clients is challenging.
+func TestStartStore(t *testing.T) {
+	// Ok and erroring subscribe functions are defined here to de-clutter tests.
+	okSubscribeFunc := func() (*subscribe.Client, error) {
+		return &subscribe.Client{}, nil
+	}
+
+	errSubscribeFunc := func() (client *subscribe.Client, e error) {
+		return nil, errors.New("intentional test err")
+	}
+
+	tests := []struct {
+		name           string
+		ChannelEvents  func() (*subscribe.Client, error)
+		PeerEvents     func() (*subscribe.Client, error)
+		GetChannels    func() ([]*channeldb.OpenChannel, error)
+		expectStartErr bool
+	}{
+		{
+			name:           "Channel events fail",
+			ChannelEvents:  errSubscribeFunc,
+			expectStartErr: true,
+		},
+		{
+			name:           "Peer events fail",
+			ChannelEvents:  okSubscribeFunc,
+			PeerEvents:     errSubscribeFunc,
+			expectStartErr: true,
+		},
+		{
+			name:          "Get oepn channels fails",
+			ChannelEvents: okSubscribeFunc,
+			PeerEvents:    okSubscribeFunc,
+			GetChannels: func() (channels []*channeldb.OpenChannel, e error) {
+				return nil, errors.New("intentional test err")
+			},
+			expectStartErr: true,
```

Comment:
add a test case for no error?

Reply:
If we don't error, the start func will try to touch the subscribe clients `updates.ChanOut`, since it's unexported it can't be mocked. Will remove that bool since and update the comment to note that.